# 03 · Payment Integrity Analysis

**Medicare Claims Payment Integrity Analytics** · Notebook 3 of 3

## Business case

In FY2025, Medicare Fee-for-Service made an estimated **$28.83 billion in improper payments**, representing **6.55% of total payments**. For Part B physician and supplier claims, the improper payment rate was **8.4%**. Approximately two-thirds of these errors were associated with claims paid without sufficient documentation to support the services billed. [CMS](https://www.cms.gov/newsroom/fact-sheets/fiscal-year-2025-improper-payments-fact-sheet) · [CMS CERT](https://www.cms.gov/files/document/nov-2025-medicare-ffs-supplemental-improper-payment-data-2025922.pdf)

Once an improper claim has been paid, recovering the payment can be slower and more costly than preventing it before payment. Because a payment integrity team cannot manually review every claim, the analysis needs to identify **which claims present the greatest payment risk and which providers warrant review priority**.

## Analytical questions

This analysis addresses six questions:

1. **Are denials increasing, and what is driving them?**
2. **Can simple prepayment rules identify high-risk claims before payment?**
3. **Which rules are appropriate for automated denial, and which require human review?**
4. **Which previously paid claims should be prioritized for recovery?**
5. **Which providers should be reviewed first?**
6. **Do targeted rules provide greater financial value than reviewing every claim?**

The analysis progresses from **historical denial patterns → prepayment rules → rule disposition → recovery opportunities → provider prioritization → financial impact**.

In [ ]:
# PostgreSQL connection (replace <your_password> before running)
%load_ext sql
%sql postgresql://postgres:password@localhost:5432/medicare_claims_db
%config SqlMagic.displaylimit = 20

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Connecting and switching to connection 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

In [4]:
%%sql
CREATE SCHEMA IF NOT EXISTS mart;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

++
||
++
++

## 1. Denial trends and drivers

**Are denials increasing, and what is driving them?**

**Why:** Before developing prepayment rules, we need a baseline view of denial activity: how many claims are denied, whether denial rates are changing over time, and which factors are associated with those denials.

### Claim-level status

The payer's 835 remittance contains multiple rows for a claim, representing individual payment actions. These records are consolidated to determine **one final status per claim**.

A reversal (`status = 22`) indicates that a claim was paid and the payment was subsequently taken back. For the payment integrity analysis, claims with a final status of **denied or reversed** are treated as **claims the payer did not ultimately pay**.

This claim-level status becomes the **benchmark for the analyses that follow**, providing the reference outcome against which prepayment rules, recovery opportunities, and provider-level risk are evaluated.

In [5]:
%%sql
DROP TABLE IF EXISTS mart.claim_outcome CASCADE;

CREATE TABLE mart.claim_outcome AS
SELECT c.clm_id,
       c.rendering_npi,
       c.hcpcs_cd,
       c.received_ts::date AS received_dt,
       CASE
           WHEN COUNT(r.remit_id) = 0             THEN 'Pending'
           WHEN BOOL_OR(r.claim_status_cd = '22') THEN 'Reversed'
           WHEN BOOL_OR(r.claim_status_cd = '4')  THEN 'Denied'
           ELSE 'Paid'
       END AS final_status,
       MAX(r.carc_cd) AS carc_cd,
       COALESCE(SUM(r.paid_amt), 0) AS net_paid
FROM core.claims c
LEFT JOIN core.remittance r ON r.clm_id = c.clm_id
GROUP BY c.clm_id, c.rendering_npi, c.hcpcs_cd, c.received_ts;

ALTER TABLE mart.claim_outcome ADD PRIMARY KEY (clm_id);

SELECT COUNT(*) AS claims FROM mart.claim_outcome;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

23980 rows affected.

1 rows affected.

claims
23980


In [6]:
%%sql
SELECT final_status,
       COUNT(*) AS claims,
       ROUND(100.0 * COUNT(*) / SUM(COUNT(*)) OVER (), 1) AS pct_of_claims,
       SUM(net_paid) AS net_paid
FROM mart.claim_outcome
GROUP BY final_status
ORDER BY claims DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

4 rows affected.

final_status,claims,pct_of_claims,net_paid
Paid,21949,91.5,1232637.04
Denied,1629,6.8,0.00
Pending,349,1.5,0
Reversed,53,0.2,0.00


In [7]:
%%sql
SELECT DATE_TRUNC('quarter', received_dt)::date AS quarter,
       COUNT(*) FILTER (WHERE final_status <> 'Pending') AS decided,
       COUNT(*) FILTER (WHERE final_status = 'Denied')   AS denied,
       ROUND(100.0 * COUNT(*) FILTER (WHERE final_status = 'Denied')
             / COUNT(*) FILTER (WHERE final_status <> 'Pending'), 1) AS denial_rate
FROM mart.claim_outcome
GROUP BY 1
ORDER BY 1;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

6 rows affected.

quarter,decided,denied,denial_rate
2025-01-01,4152,240,5.8
2025-04-01,4049,276,6.8
2025-07-01,3681,219,5.9
2025-10-01,4022,260,6.5
2026-01-01,4076,333,8.2
2026-04-01,3651,301,8.2


### Finding

The denial rate remained between **5.8% and 6.8% throughout 2025**, then increased to **8.2% in both quarters of 2026**. The increase persisted across two consecutive quarters rather than appearing as a one-quarter spike.

### Insight

The sustained increase suggests a **change in the underlying denial pattern going into 2026**, rather than a temporary fluctuation. The next analysis should determine whether the shift is concentrated in specific services, denial reasons, or providers.

**Next:** Break denials down by **service and denial reason** to identify the primary drivers of the increase.

In [8]:
%%sql
SELECT o.hcpcs_cd,
       f.short_description,
       COUNT(*) FILTER (WHERE o.final_status <> 'Pending') AS decided,
       COUNT(*) FILTER (WHERE o.final_status = 'Denied')   AS denied,
       ROUND(100.0 * COUNT(*) FILTER (WHERE o.final_status = 'Denied')
             / COUNT(*) FILTER (WHERE o.final_status <> 'Pending'), 1) AS denial_rate,
       SUM(o.net_paid) AS net_paid
FROM mart.claim_outcome o
JOIN core.fee_schedule f
  ON f.hcpcs_cd = o.hcpcs_cd AND f.effective_start_dt = '2025-01-01'
GROUP BY o.hcpcs_cd, f.short_description
ORDER BY denial_rate DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

5 rows affected.

hcpcs_cd,short_description,decided,denied,denial_rate,net_paid
97110,"Therapeutic exercise, each 15 minutes",5956,693,11.6,234469.96
80053,Comprehensive metabolic panel (lab),1694,122,7.2,16359.82
93000,"Electrocardiogram, 12-lead, with interpretation and report",1947,136,7.0,17448.68
99213,"Established patient office visit, low complexity",6480,316,4.9,356108.77
99214,"Established patient office visit, moderate complexity",7554,362,4.8,608249.81


In [9]:
%%sql
SELECT rc.category AS denial_reason,
       COUNT(*) AS denied_claims,
       ROUND(100.0 * COUNT(*) / SUM(COUNT(*)) OVER (), 1) AS pct_of_denials
FROM mart.claim_outcome o
JOIN core.reason_codes rc
  ON rc.code = o.carc_cd AND rc.code_set = 'CARC'
WHERE o.final_status = 'Denied'
GROUP BY rc.category
ORDER BY denied_claims DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

8 rows affected.

denial_reason,denied_claims,pct_of_denials
missing_information,441,27.1
medical_necessity,378,23.2
duplicate,234,14.4
eligibility,150,9.2
coding,142,8.7
units_frequency,109,6.7
provider_enrollment,89,5.5
timely_filing,86,5.3


To see what drove the 2026 increase, denials are normalized **per 1,000 decided claims**, since 2026 only covers two quarters.

In [10]:
%%sql
WITH decided AS (
    SELECT EXTRACT(YEAR FROM received_dt) AS yr, COUNT(*) AS n
    FROM mart.claim_outcome
    WHERE final_status <> 'Pending'
    GROUP BY 1
),
denials AS (
    SELECT rc.category,
           EXTRACT(YEAR FROM o.received_dt) AS yr,
           COUNT(*) AS denied
    FROM mart.claim_outcome o
    JOIN core.reason_codes rc
      ON rc.code = o.carc_cd AND rc.code_set = 'CARC'
    WHERE o.final_status = 'Denied'
    GROUP BY 1, 2
)
SELECT d.category AS denial_reason,
       ROUND(1000.0 * SUM(d.denied) FILTER (WHERE d.yr = 2025) / MAX(c.n) FILTER (WHERE c.yr = 2025), 1) AS per_1k_2025,
       ROUND(1000.0 * SUM(d.denied) FILTER (WHERE d.yr = 2026) / MAX(c.n) FILTER (WHERE c.yr = 2026), 1) AS per_1k_2026
FROM denials d
JOIN decided c ON c.yr = d.yr
GROUP BY d.category
ORDER BY 3 DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

8 rows affected.

denial_reason,per_1k_2025,per_1k_2026
missing_information,17.4,21.2
medical_necessity,14.0,20.1
duplicate,9.1,11.5
eligibility,6.0,7.0
provider_enrollment,2.3,6.7
coding,5.8,6.5
timely_filing,3.2,4.5
units_frequency,4.7,4.5


### Findings

- **97110 (Therapeutic Exercise)** has the highest denial rate at **11.6%**, more than twice the rate of office-visit codes, while also ranking third by dollars paid.
- **Missing information and medical necessity** account for **50% of all denials**. Both point to **incomplete claims or documentation**.
- The 2026 increase is broad: **7 of 8 denial reasons increased**. Medical necessity rose the most (**+6.1 per 1,000**), followed by provider enrollment (**+4.4**) and missing information (**+3.8**).
- **Provider enrollment** denials nearly tripled, from **2.3 to 6.7 per 1,000 decided claims**.

### Insights

- **97110 is the highest-risk service line** by denial rate and should be a priority when developing prepayment edits.
- **Incomplete claims and documentation are the largest denial drivers**, making medical necessity and missing-information checks important areas for rule development.
- The rise in **provider enrollment denials** suggests a potential prepayment control opportunity. Validating provider enrollment and effective dates before payment could identify these claims earlier.

**Next:** Build prepayment edits using claim, eligibility, enrollment, and fee schedule data, then test how many observed denials the rules could have caught before payment.

## 2. Prepayment rule testing

**Can simple prepayment rules catch potentially improper claims before they are paid?**

**Why:** If rules using only information available when a claim arrives can reproduce payer decisions, they can screen claims before payment and route exceptions to reviewers.

Each rule, or **prepayment edit**, is based on a published Medicare billing requirement and mapped to the payer denial reason it is designed to replicate. [X12 CARC Codes](https://x12.org/codes/claim-adjustment-reason-codes)

| Edit | Rule | Payer reason it mirrors (CARC) |
|---|---|---|
| `not_covered` | Service date outside patient's Part B coverage | 24, 26, 27 (Eligibility) |
| `after_death` | Service date after patient's date of death | 27 (Eligibility) |
| `not_enrolled` | Service date outside provider's Medicare enrollment | B7 (Provider enrollment) |
| `late_filing` | Claim received more than 1 year after service | 29 (Timely filing) |
| `over_units` | Units above the fee schedule's daily maximum | 151 (Units/frequency) |
| `missing_referring` | Lab or therapy claim with no ordering/referring NPI | 16 (Missing information) |
| `missing_dx` | No ICD-10-CM diagnosis code | 16 (Missing information) |
| `bad_modifier` | Therapy without GP modifier, or GP/59 on a non-therapy code | 4 (Coding) |
| `screening_dx` | Screening diagnosis (Z00, Z13) on any service, or a diagnosis that does not support therapy (97110 billed with I10 or J069) | 50 (Medical necessity) |
| `duplicate` | Same patient, provider, code, date, units, and modifier already billed and not denied | 18 (Duplicate) |

In [11]:
%%sql
-- CASCADE also drops the view mart.claim_rule_flags (used by Power BI); it is recreated at the end of this cell
DROP TABLE IF EXISTS mart.claim_checks CASCADE;

CREATE TABLE mart.claim_checks AS
SELECT c.clm_id,
       c.rendering_npi,
       c.hcpcs_cd,
       o.final_status,
       o.net_paid,
       f.allowed_amount_per_unit * c.units AS expected_payment,

       (c.service_dt < p.ptb_start_dt OR c.service_dt > p.ptb_end_dt)        AS not_covered,
       (c.service_dt > p.death_dt)                                            AS after_death,
       (c.service_dt < v.enroll_start_dt OR c.service_dt > v.enroll_end_dt)   AS not_enrolled,
       (c.received_ts > c.service_dt + INTERVAL '1 year')                     AS late_filing,
       (c.units > f.max_units_per_day)                                        AS over_units,
       (c.hcpcs_cd IN ('97110', '80053') AND c.referring_npi IS NULL)         AS missing_referring,
       (c.dx_cd IS NULL)                                                      AS missing_dx,
       ((c.hcpcs_cd = '97110' AND c.modifier IS DISTINCT FROM 'GP')
         OR (c.hcpcs_cd <> '97110' AND c.modifier IN ('GP', '59')))           AS bad_modifier,
       (c.dx_cd LIKE 'Z00%' OR c.dx_cd LIKE 'Z13%'
         OR (c.hcpcs_cd = '97110' AND c.dx_cd IN ('I10', 'J069')))            AS screening_dx,
       EXISTS (
           SELECT 1
           FROM core.claims d
           JOIN mart.claim_outcome od ON od.clm_id = d.clm_id
           WHERE d.bene_id = c.bene_id
             AND d.rendering_npi = c.rendering_npi
             AND d.hcpcs_cd = c.hcpcs_cd
             AND d.service_dt = c.service_dt
             AND d.units = c.units
             AND d.modifier IS NOT DISTINCT FROM c.modifier
             AND d.received_ts < c.received_ts
             AND od.final_status <> 'Denied'
       ) AS duplicate

FROM core.claims c
JOIN mart.claim_outcome o  ON o.clm_id = c.clm_id
LEFT JOIN core.patients p  ON p.bene_id = c.bene_id
LEFT JOIN core.providers v ON v.npi = c.rendering_npi
LEFT JOIN core.fee_schedule f
       ON f.hcpcs_cd = c.hcpcs_cd
      AND c.service_dt BETWEEN f.effective_start_dt AND f.effective_end_dt;

ALTER TABLE mart.claim_checks ADD PRIMARY KEY (clm_id);

-- One row per claim per failed edit, for the Power BI report
CREATE OR REPLACE VIEW mart.claim_rule_flags AS
SELECT k.clm_id, r.rule_order, r.rule_name
FROM mart.claim_checks k
CROSS JOIN LATERAL (VALUES
    (1,  'Service after death',        k.after_death),
    (2,  'Late filing (over 1 year)',  k.late_filing),
    (3,  'No Part B coverage',         k.not_covered),
    (4,  'Provider not enrolled',      k.not_enrolled),
    (5,  'Duplicate claim',            k.duplicate),
    (6,  'Units over daily limit',     k.over_units),
    (7,  'Missing diagnosis',          k.missing_dx),
    (8,  'Wrong modifier',             k.bad_modifier),
    (9,  'Missing referring provider', k.missing_referring),
    (10, 'Screening dx on diagnostic', k.screening_dx)
) AS r (rule_order, rule_name, failed)
WHERE r.failed;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

23980 rows affected.

++
||
++
++

A claim is **flagged** when **at least one edit fires**. A `NULL` edit result, such as a missing date, is treated as **not fired**.

In [12]:
%%sql
ALTER TABLE mart.claim_checks ADD COLUMN flagged boolean;

UPDATE mart.claim_checks
SET flagged = (not_covered OR after_death OR not_enrolled OR late_filing OR over_units
               OR missing_referring OR missing_dx OR bad_modifier OR screening_dx
               OR duplicate) IS TRUE;

SELECT COUNT(*) FILTER (WHERE flagged) AS flagged_claims,
       ROUND(100.0 * COUNT(*) FILTER (WHERE flagged) / COUNT(*), 1) AS pct_flagged
FROM mart.claim_checks;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

23980 rows affected.

1 rows affected.

flagged_claims,pct_flagged
1497,6.2


### Rule evaluation

We compare the rule flags against the payer's final decision using a **confusion matrix**.

- **Precision:** Of the claims we flag, the share the payer also refused to pay.
- **Recall:** Of the claims the payer refused to pay, the share we successfully flagged.

In [13]:
%%sql
SELECT COUNT(*) FILTER (WHERE flagged AND final_status IN ('Denied', 'Reversed'))     AS true_positive,
       COUNT(*) FILTER (WHERE flagged AND final_status = 'Paid')                      AS false_positive,
       COUNT(*) FILTER (WHERE NOT flagged AND final_status IN ('Denied', 'Reversed')) AS false_negative,
       COUNT(*) FILTER (WHERE NOT flagged AND final_status = 'Paid')                  AS true_negative,
       ROUND(100.0 * COUNT(*) FILTER (WHERE flagged AND final_status IN ('Denied', 'Reversed'))
             / COUNT(*) FILTER (WHERE flagged), 1) AS precision_pct,
       ROUND(100.0 * COUNT(*) FILTER (WHERE flagged AND final_status IN ('Denied', 'Reversed'))
             / COUNT(*) FILTER (WHERE final_status IN ('Denied', 'Reversed')), 1) AS recall_pct
FROM mart.claim_checks
WHERE final_status <> 'Pending';

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

1 rows affected.

true_positive,false_positive,false_negative,true_negative,precision_pct,recall_pct
1386,99,296,21850,93.3,82.4


Which denials do the edits miss?

In [14]:
%%sql
SELECT rc.category AS denial_reason,
       COUNT(*) AS denied,
       COUNT(*) FILTER (WHERE k.flagged) AS caught,
       COUNT(*) FILTER (WHERE NOT k.flagged) AS missed,
       ROUND(100.0 * COUNT(*) FILTER (WHERE k.flagged) / COUNT(*), 1) AS recall_pct
FROM mart.claim_checks k
JOIN mart.claim_outcome o ON o.clm_id = k.clm_id
JOIN core.reason_codes rc ON rc.code = o.carc_cd AND rc.code_set = 'CARC'
WHERE k.final_status = 'Denied'
GROUP BY rc.category
ORDER BY recall_pct;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

8 rows affected.

denial_reason,denied,caught,missed,recall_pct
medical_necessity,378,227,151,60.1
missing_information,441,297,144,67.3
timely_filing,86,85,1,98.8
provider_enrollment,89,89,0,100.0
coding,142,142,0,100.0
units_frequency,109,109,0,100.0
duplicate,234,234,0,100.0
eligibility,150,150,0,100.0


### Findings

- Edits flag **6.2% of all claims**, with **93.3% precision** and **82.4% recall**.
- They catch **100%** of duplicate, eligibility, coding, units, and provider-enrollment denials.
- Of the **296 missed denials**, **151 were medical necessity** and **144 were missing information**.

### Insight

The rules perform well on denials that can be identified from **claim, eligibility, enrollment, and billing data**. Most missed denials require **supporting medical documentation**, which is not available in the claim data used for these edits.

This aligns with CMS's FY2025 finding that **insufficient or missing documentation accounted for 65% of Medicare FFS improper payments**. [CMS CERT Report](https://www.cms.gov/files/document/nov-2025-medicare-ffs-supplemental-improper-payment-data-2025922.pdf)

**Next:** Measure precision for each edit to identify rules generating the most false positives.

## 3. Rule disposition: auto-deny vs. manual review

**Which rules are safe to auto-deny, and which should go to a reviewer?**

**Why:** An auto-deny rule needs to be highly accurate to avoid unnecessary provider appeals. Lower-precision rules can still be useful as review signals.

**Decision Rule:** As a **business assumption**, edits with **≥98% precision** are classified as **hard edits** and can auto-deny. Edits below 98% are classified as **soft edits** and route claims to manual review.

In [15]:
%%sql
SELECT e.edit,
       COUNT(*) AS flagged,
       COUNT(*) FILTER (WHERE k.final_status IN ('Denied', 'Reversed')) AS payer_refused,
       COUNT(*) FILTER (WHERE k.final_status = 'Paid') AS payer_paid,
       ROUND(100.0 * COUNT(*) FILTER (WHERE k.final_status IN ('Denied', 'Reversed')) / COUNT(*), 1) AS precision_pct,
       CASE
           WHEN 100.0 * COUNT(*) FILTER (WHERE k.final_status IN ('Denied', 'Reversed')) / COUNT(*) >= 98
           THEN 'Hard edit: auto-deny'
           ELSE 'Soft edit: route to review'
       END AS recommended_action
FROM mart.claim_checks k
CROSS JOIN LATERAL (VALUES
    ('not_covered',       k.not_covered),
    ('after_death',       k.after_death),
    ('not_enrolled',      k.not_enrolled),
    ('late_filing',       k.late_filing),
    ('over_units',        k.over_units),
    ('missing_referring', k.missing_referring),
    ('missing_dx',        k.missing_dx),
    ('bad_modifier',      k.bad_modifier),
    ('screening_dx',      k.screening_dx),
    ('duplicate',         k.duplicate)
) AS e(edit, hit)
WHERE e.hit
  AND k.final_status <> 'Pending'
GROUP BY e.edit
ORDER BY precision_pct DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

10 rows affected.

edit,flagged,payer_refused,payer_paid,precision_pct,recommended_action
missing_referring,193,193,0,100.0,Hard edit: auto-deny
after_death,34,34,0,100.0,Hard edit: auto-deny
missing_dx,115,115,0,100.0,Hard edit: auto-deny
late_filing,85,85,0,100.0,Hard edit: auto-deny
not_enrolled,92,91,1,98.9,Hard edit: auto-deny
not_covered,156,153,3,98.1,Hard edit: auto-deny
duplicate,269,250,19,92.9,Soft edit: route to review
bad_modifier,169,156,13,92.3,Soft edit: route to review
over_units,142,127,15,89.4,Soft edit: route to review
screening_dx,325,276,49,84.9,Soft edit: route to review


### Findings

- **6 edits meet the ≥98% precision threshold** and qualify as hard edits:
  - Missing referring NPI
  - Missing diagnosis
  - After death
  - Late filing
  - Provider not enrolled
  - Coverage
- The remaining **4 edits** range from **84.9% to 92.9% precision** and should route claims to manual review.
- **Screening diagnosis** is the weakest rule, with **49 flagged claims that the payer paid**.

### Insight

Most rules meet the threshold for **safe auto-denial**, while the four lower-precision rules are better suited for reviewer assessment. The **screening diagnosis** rule needs a tighter ICD-10-CM code list before production use.

> **Precision caveat:** "Payer paid" does not necessarily mean the claim was appropriate. Some paid claims may represent improper payments, so the measured precision is **likely conservative**. These paid-but-flagged claims are examined as recovery opportunities.

## 4. Recovery opportunities

**Which paid claims should we try to recover?**

**Why:** A paid claim that fails a prepayment edit represents a **potential overpayment** and becomes a candidate for post-payment review and recovery.

In [16]:
%%sql
SELECT COUNT(*) AS recovery_candidates,
       SUM(net_paid) AS paid_amt,
       ROUND(100.0 * SUM(net_paid) / (SELECT SUM(net_paid) FROM mart.claim_outcome), 2) AS pct_of_total_paid
FROM mart.claim_checks
WHERE flagged AND final_status = 'Paid';

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

1 rows affected.

recovery_candidates,paid_amt,pct_of_total_paid
99,7088.10,0.58


Each claim is assigned **one primary reason**, based on the first edit that fires in the predefined priority order, with the most objective edits evaluated first.

In [17]:
%%sql
SELECT reason,
       COUNT(*) AS claims,
       SUM(net_paid) AS paid_amt,
       ROUND(AVG(net_paid), 2) AS avg_paid
FROM (
    SELECT net_paid,
           CASE
               WHEN not_covered  THEN 'Not covered'
               WHEN not_enrolled THEN 'Provider not enrolled'
               WHEN duplicate    THEN 'Duplicate'
               WHEN over_units   THEN 'Over unit limit'
               WHEN bad_modifier THEN 'Wrong modifier'
               WHEN screening_dx THEN 'Screening diagnosis'
           END AS reason
    FROM mart.claim_checks
    WHERE flagged AND final_status = 'Paid'
) x
GROUP BY reason
ORDER BY paid_amt DESC;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

6 rows affected.

reason,claims,paid_amt,avg_paid
Screening diagnosis,48,2761.38,57.53
Over unit limit,15,2126.82,141.79
Duplicate,19,1292.37,68.02
Wrong modifier,13,725.07,55.77
Not covered,3,125.09,41.70
Provider not enrolled,1,57.37,57.37


### Recovery worklist

Top recovery candidates, ranked by **dollars at risk**:

In [18]:
%%sql
SELECT clm_id,
       rendering_npi,
       hcpcs_cd,
       net_paid,
       CASE
           WHEN not_covered  THEN 'Not covered'
           WHEN not_enrolled THEN 'Provider not enrolled'
           WHEN duplicate    THEN 'Duplicate'
           WHEN over_units   THEN 'Over unit limit'
           WHEN bad_modifier THEN 'Wrong modifier'
           WHEN screening_dx THEN 'Screening diagnosis'
       END AS reason
FROM mart.claim_checks
WHERE flagged AND final_status = 'Paid'
ORDER BY net_paid DESC
LIMIT 10;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

10 rows affected.

clm_id,rendering_npi,hcpcs_cd,net_paid,reason
542100021728,9527877796,99213,215.11,Over unit limit
542100023254,9349942661,99214,203.40,Over unit limit
542100002703,9349942661,99214,197.47,Over unit limit
542100003332,9309927900,97110,191.22,Over unit limit
542100017208,9097504242,97110,191.22,Over unit limit
542100021194,9903502281,97110,177.24,Over unit limit
542100005846,9640705353,97110,172.10,Over unit limit
542100017981,9697743257,97110,157.55,Over unit limit
542100016664,9746951158,97110,152.98,Over unit limit
542100018058,9504420982,99213,143.41,Over unit limit


### Findings

- **99 paid claims** fail at least one edit, totaling **$7,088 in paid amounts at risk**, or **0.6% of total dollars paid**.
- **Screening diagnosis** has the most flagged claims: **48 claims / $2,761**.
- **Over unit limit** has the highest average amount at risk: **$142 per claim**.
- The **top 10 claims by dollars at risk** are all over-unit cases: **6 therapy (97110)** claims and **4 office-visit** claims.

### Insight

These claims are **recovery leads, not confirmed overpayments**. Because some edits produce false positives, each claim requires medical-record review before recovery action.

The $7,088 is the **full paid amount** on these claims, so it is an **upper bound**. For over-unit claims, the recoverable amount is usually only the payment for the units above the limit.

The worklist should therefore be prioritized by **dollars at risk**, starting with the over-unit claims at the top of the list.

**Next:** Move from individual claims to **provider-level patterns** to identify providers who may require outreach or further review.

## 5. Provider review prioritization

**Which providers should we review first?**

**Why:** CMS's [Targeted Probe and Educate (TPE) program](https://www.cms.gov/Research-Statistics-Data-and-Systems/Monitoring-Programs/Medicare-FFS-Compliance-Programs/Medical-Review/Targeted-Probe-and-EducateTPE.html) reviews a sample of claims from selected providers based on claim error rates. Accurate provider prioritization helps focus limited review capacity where the risk is highest.

**Metric:** **Provider flag rate** = flagged claims ÷ total claims.

In [19]:
%%sql
DROP TABLE IF EXISTS mart.provider_risk;

CREATE TABLE mart.provider_risk AS
SELECT k.rendering_npi,
       v.last_name,
       v.specialty,
       COUNT(*) AS claims,
       COUNT(*) FILTER (WHERE k.flagged) AS flagged,
       ROUND(100.0 * COUNT(*) FILTER (WHERE k.flagged) / COUNT(*), 1) AS flag_rate,
       NTILE(4) OVER (ORDER BY COUNT(*) FILTER (WHERE k.flagged)::numeric / COUNT(*) DESC) AS risk_quartile
FROM mart.claim_checks k
JOIN core.providers v ON v.npi = k.rendering_npi
GROUP BY k.rendering_npi, v.last_name, v.specialty;

ALTER TABLE mart.provider_risk ADD PRIMARY KEY (rendering_npi);

SELECT COUNT(*) AS providers FROM mart.provider_risk;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

120 rows affected.

1 rows affected.

providers
120


In [20]:
%%sql
SELECT RANK() OVER (ORDER BY flag_rate DESC) AS risk_rank,
       rendering_npi,
       last_name,
       specialty,
       claims,
       flagged,
       flag_rate
FROM mart.provider_risk
ORDER BY risk_rank
LIMIT 10;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

10 rows affected.

risk_rank,rendering_npi,last_name,specialty,claims,flagged,flag_rate
1,9251274384,Jaramillo,Internal Medicine,51,16,31.4
2,9309927900,Galloway,Physical Therapist,467,129,27.6
3,9097504242,Haddad,Physical Therapist,290,79,27.2
4,9746951158,Gupta,Physical Therapist,284,66,23.2
5,9640705353,Moreno,Physical Therapist,271,58,21.4
6,9299902954,Valdez,Internal Medicine,100,17,17.0
7,9739303912,Dabrowski,Family Medicine,47,7,14.9
8,9583660094,Brennan,Family Medicine,115,17,14.8
9,9814243462,Delacroix,Internal Medicine,222,29,13.1
10,9189919084,Yilmaz,Family Medicine,136,17,12.5


In [21]:
%%sql

-- How concentrated are the errors?

SELECT risk_quartile,
       COUNT(*) AS providers,
       SUM(claims) AS claims,
       SUM(flagged) AS flagged,
       ROUND(100.0 * SUM(flagged) / SUM(claims), 1) AS flag_rate,
       ROUND(100.0 * SUM(claims) / SUM(SUM(claims)) OVER (), 1) AS pct_of_claims,
       ROUND(100.0 * SUM(flagged) / SUM(SUM(flagged)) OVER (), 1) AS pct_of_flagged
FROM mart.provider_risk
GROUP BY risk_quartile
ORDER BY risk_quartile;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

4 rows affected.

risk_quartile,providers,claims,flagged,flag_rate,pct_of_claims,pct_of_flagged
1,30,7715,944,12.2,32.2,63.1
2,30,5960,288,4.8,24.9,19.2
3,30,4964,163,3.3,20.7,10.9
4,30,5341,102,1.9,22.3,6.8


### Findings

- The **top risk quartile (30 providers)** submits **32% of claims** but accounts for **63.1% of flagged claims**.
- Their **12.2% flag rate** is **6.4×** the bottom quartile's **1.9%**.
- **4 of the top 5 providers are physical therapists**, consistent with the higher denial rate observed for 97110 in Section 1.

### Insight

Errors are **highly concentrated among a small group of providers**. A review program focused on the top 30 providers could therefore reach nearly **two-thirds of flagged claims** while covering only 32% of claim volume.

However, the **#1 provider has only 51 claims**, so the high flag rate should be validated with a claim sample before initiating provider outreach.

**Next:** Evaluate whether the **rules-plus-review model** provides greater financial value than reviewing every claim.

## 6. Cost trade-off analysis

**Do the rules save money compared with reviewing every claim?**

**Why:** Leadership needs to understand the cost trade-off between reviewing every claim and using automated edits. Rules reduce manual review volume but may allow some denied claims to be paid.

### Scenario assumptions

*Illustrative assumptions, not measured costs.*

| Cost driver | Base case |
|---|---:|
| Manual review, per claim | $18 |
| Automated edit processing, per claim | $0.35 |
| Wrong approval | Expected payment + $250 rework/recovery cost |
| Manual review accuracy | Catches every claim the payer refused (no wrong approvals) |

### Models

- **Manual model:** Every decided claim is manually reviewed, with no wrong approvals.
- **Rules-based model:** Claims that fail a hard edit (Section 3) are **denied automatically** at the automated processing cost. Claims flagged only by soft edits go to **manual review**. Unflagged claims are **auto-processed**, and the cost of the improper claims they let through (wrong approvals) is added.

In [22]:
%%sql
WITH flags AS (
    SELECT k.clm_id, e.edit
    FROM mart.claim_checks k
    CROSS JOIN LATERAL (VALUES
    ('not_covered',       k.not_covered),
    ('after_death',       k.after_death),
    ('not_enrolled',      k.not_enrolled),
    ('late_filing',       k.late_filing),
    ('over_units',        k.over_units),
    ('missing_referring', k.missing_referring),
    ('missing_dx',        k.missing_dx),
    ('bad_modifier',      k.bad_modifier),
    ('screening_dx',      k.screening_dx),
    ('duplicate',         k.duplicate)
    ) AS e(edit, hit)
    WHERE e.hit
      AND k.final_status <> 'Pending'
),
hard_edits AS (                      -- edits with >= 98% precision auto-deny (Section 3)
    SELECT f.edit
    FROM flags f
    JOIN mart.claim_checks k USING (clm_id)
    GROUP BY f.edit
    HAVING 100.0 * COUNT(*) FILTER (WHERE k.final_status IN ('Denied', 'Reversed')) / COUNT(*) >= 98
),
auto_denied AS (                     -- claims failing at least one hard edit
    SELECT DISTINCT f.clm_id
    FROM flags f
    JOIN hard_edits h USING (edit)
),
base AS (
    SELECT COUNT(*) AS decided,
           COUNT(*) FILTER (WHERE k.flagged AND a.clm_id IS NOT NULL) AS auto_denied,
           COUNT(*) FILTER (WHERE k.flagged AND a.clm_id IS NULL) AS manual_review,
           COUNT(*) FILTER (WHERE NOT k.flagged) AS auto_paid,
           COUNT(*) FILTER (WHERE NOT k.flagged AND k.final_status IN ('Denied', 'Reversed')) AS wrong_approvals,
           SUM(COALESCE(k.expected_payment, 0))
               FILTER (WHERE NOT k.flagged AND k.final_status IN ('Denied', 'Reversed')) AS missed_payment
    FROM mart.claim_checks k
    LEFT JOIN auto_denied a USING (clm_id)
    WHERE k.final_status <> 'Pending'
)
SELECT decided,
       auto_denied,
       manual_review,
       wrong_approvals,
       decided * 18 AS manual_cost,
       ROUND(manual_review * 18 + (auto_paid + auto_denied) * 0.35
             + wrong_approvals * 250 + missed_payment) AS rules_cost,
       ROUND(decided * 18 - (manual_review * 18 + (auto_paid + auto_denied) * 0.35
             + wrong_approvals * 250 + missed_payment)) AS savings,
       ROUND(100.0 * (1 - (manual_review * 18 + (auto_paid + auto_denied) * 0.35
             + wrong_approvals * 250 + missed_payment) / (decided * 18)), 1) AS savings_pct
FROM base;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

1 rows affected.

decided,auto_denied,manual_review,wrong_approvals,manual_cost,rules_cost,savings,savings_pct
23631,631,854,296,425358,120274,305084,71.7


### Sensitivity

The cost comparison depends on two assumptions: **manual review cost** and **cost of a wrong approval**.

`breakeven_error_cost` is the **rework/recovery cost per wrong approval, on top of the claim's expected payment, at which the manual-review and rules-based models have the same total cost**. Above this point, reviewing every claim becomes cheaper.

In [23]:
%%sql
WITH flags AS (
    SELECT k.clm_id, e.edit
    FROM mart.claim_checks k
    CROSS JOIN LATERAL (VALUES
    ('not_covered',       k.not_covered),
    ('after_death',       k.after_death),
    ('not_enrolled',      k.not_enrolled),
    ('late_filing',       k.late_filing),
    ('over_units',        k.over_units),
    ('missing_referring', k.missing_referring),
    ('missing_dx',        k.missing_dx),
    ('bad_modifier',      k.bad_modifier),
    ('screening_dx',      k.screening_dx),
    ('duplicate',         k.duplicate)
    ) AS e(edit, hit)
    WHERE e.hit
      AND k.final_status <> 'Pending'
),
hard_edits AS (                      -- edits with >= 98% precision auto-deny (Section 3)
    SELECT f.edit
    FROM flags f
    JOIN mart.claim_checks k USING (clm_id)
    GROUP BY f.edit
    HAVING 100.0 * COUNT(*) FILTER (WHERE k.final_status IN ('Denied', 'Reversed')) / COUNT(*) >= 98
),
auto_denied AS (                     -- claims failing at least one hard edit
    SELECT DISTINCT f.clm_id
    FROM flags f
    JOIN hard_edits h USING (edit)
),
base AS (
    SELECT COUNT(*) AS decided,
           COUNT(*) FILTER (WHERE k.flagged AND a.clm_id IS NOT NULL) AS auto_denied,
           COUNT(*) FILTER (WHERE k.flagged AND a.clm_id IS NULL) AS manual_review,
           COUNT(*) FILTER (WHERE NOT k.flagged) AS auto_paid,
           COUNT(*) FILTER (WHERE NOT k.flagged AND k.final_status IN ('Denied', 'Reversed')) AS wrong_approvals,
           SUM(COALESCE(k.expected_payment, 0))
               FILTER (WHERE NOT k.flagged AND k.final_status IN ('Denied', 'Reversed')) AS missed_payment
    FROM mart.claim_checks k
    LEFT JOIN auto_denied a USING (clm_id)
    WHERE k.final_status <> 'Pending'
)
SELECT s.review_cost,
       s.error_cost,
       b.decided * s.review_cost AS manual_cost,
       ROUND(b.manual_review * s.review_cost + (b.auto_paid + b.auto_denied) * 0.35
             + b.wrong_approvals * s.error_cost + b.missed_payment) AS rules_cost,
       ROUND((b.decided * s.review_cost - b.manual_review * s.review_cost
              - (b.auto_paid + b.auto_denied) * 0.35 - b.missed_payment) / b.wrong_approvals) AS breakeven_error_cost
FROM base b
CROSS JOIN (VALUES (8, 250), (18, 250), (35, 250), (8, 2000), (18, 2000), (35, 2000))
     AS s(review_cost, error_cost)
ORDER BY s.error_cost, s.review_cost;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

6 rows affected.

review_cost,error_cost,manual_cost,rules_cost,breakeven_error_cost
8,250,189048,111734,511
18,250,425358,120274,1281
35,250,827085,134792,2589
8,2000,189048,629734,511
18,2000,425358,638274,1281
35,2000,827085,652792,2589


### Findings

- **Base case:** Rules cost **$120,274** vs. **$425,358** for full manual review, **72% lower** (**$305,084** saved). The 631 claims that fail a hard edit are denied automatically, so only 854 flagged claims need a reviewer.
- Rules stay cheaper until the **rework/recovery cost per wrong approval** exceeds **$1,281** at an **$18 review cost**.
- At **$2,000 rework/recovery cost per wrong approval**, full manual review is cheaper at both **$8 and $18 per-review costs**.

### Insight

The economic advantage of automation depends heavily on the **cost of missed claims**, while higher manual-review costs make the rules-based model more attractive.

Because most missed denials are **medical necessity and missing documentation**, the better approach is to **use rules for broad screening and routing, then target medical-record review to the highest-risk segments**, particularly 97110 therapy claims and top-quartile providers.

## Summary of findings

| # | Finding | Recommended action |
|---|---|---|
| 1 | Denial rate increased to **8.2% in both 2026 quarters**. 97110 therapy has the highest denial rate (**11.6%**), incomplete claims and documentation drive half of denials, and enrollment denials nearly tripled. | Prioritize therapy claims and add an enrollment-date check at claim intake. |
| 2 | Rules achieve **93.3% precision** and **82.4% recall**; most misses are medical necessity and missing information. | Use rules for routing; use medical-record review for documentation-related denials. |
| 3 | **6 of 10 rules** achieve at least **98% precision**. | Deploy 6 as hard edits and 4 as soft edits; tighten the screening diagnosis code list. |
| 4 | **99 paid claims ($7,088)** fail at least one rule. | Request records in dollar order, starting with over-unit claims. |
| 5 | The **top provider quartile accounts for 63.1% of flagged claims**. | Prioritize top-quartile providers for TPE review; confirm high flag rates on low-volume providers with a claim sample before outreach. |
| 6 | Rules cost **72% less** in the base case ($305,084 saved). They stay cheaper until the rework/recovery cost per wrong approval exceeds **$1,281**. | Use rules-based routing with targeted medical-record review for high-risk segments. |

### Limitations

- The data is **synthetic and modeled on CMS file layouts**; results demonstrate the analytical method rather than real Medicare outcomes.
- The payer's decision is the benchmark, so payer errors count against the edits. **Measured precision is therefore likely conservative.**
- Cost inputs are **scenario assumptions**, which is why sensitivity analysis is included.
- **349 pending claims** are excluded from denial rates, precision and recall, and the cost model. Flag rates in Sections 2 and 5 include them.

In [24]:
%%sql
SELECT table_name
FROM information_schema.tables
WHERE table_schema = 'mart'
ORDER BY table_name;

Running query in 'postgresql://postgres:***@localhost:5432/medicare_claims_db'

4 rows affected.

table_name
claim_checks
claim_outcome
claim_rule_flags
provider_risk


The `mart` tables are ready for the Power BI layer.